### PR TITLE
Fixed PyTorch 1.1.0 use, Seeding, Refactoring

### PR DESCRIPTION
**Pull request by GerardMaggiolino**
In reference to issue #42. 

Changes to fix PyTorch 1.1.0 incompatibilities, add seeding functionality,
deleting unused lines.
- Change of attributes and added context manager use in data.load_data and
  model.Ganomaly.set_input
- Removed unused statements and redundant attributes throughout model.Ganomaly
- Added previously missing seeding functionality to data loading, model
  initialization, and training procedure based on opt args.

Please let me know if these changes look good, or if any change of implementation would be preferred before merging. Thanks!

### CODE REVIEW

**Comment by samet-akcay** [comment]
Thanks, @GerardMaggiolino. I'll have a look today, and merge it.

**Comment by GerardMaggiolino** [comment]
I’ve actually found one bug on one line while using this PR’d code; when using droplast=False for the training dataloader. This can be fixed by resizing the labels, without calling .data before resizing on line 165 of model.py. I’ll fix this and squash it into the commit.

**Comment by GerardMaggiolino** [comment]
Probably not the best form, but I fixed that issue and force pushed the change to keep it under one commit 🙂. Let me know if it works.
